operator/pkg/rest/getter.go

```diff
@@ -1,26 +1,28 @@
 package rest
 
 import (
-	"sync"
-
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/client-go/discovery"
 	memory "k8s.io/client-go/discovery/cached"
 	"k8s.io/client-go/rest"
 	"k8s.io/client-go/restmapper"
 	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/module-manager/operator/pkg/types"
 )
 
 type ManifestRESTClientGetter struct {
-	config          *rest.Config
-	client          discovery.CachedDiscoveryInterface
-	clientSyncMutex sync.Mutex
+	config     *rest.Config
+	cache      types.HelmClientCache
+	kymaObjKey client.ObjectKey
 }
 
-func NewRESTClientGetter(config *rest.Config) *ManifestRESTClientGetter {
+func NewRESTClientGetter(config *rest.Config, cache types.HelmClientCache, kymaObjKey client.ObjectKey) *ManifestRESTClientGetter {
 	return &ManifestRESTClientGetter{
-		config:          config,
-		clientSyncMutex: sync.Mutex{},
+		config:     config,
+		cache:      cache,
+		kymaObjKey: kymaObjKey,
```

Comment:
you do not need this key since there will be at most one getter for every kyma / helm client / cluster anyways.

Reply:
See https://github.com/kyma-project/module-manager/pull/145/commits/15f0a41da83694c714184a70244e9bef95ede6e7#diff-a8e4298e04e7b2a5e7e1e9c4c6187ba810e038698ab91c53f98ed3b802968987R145-R146

Reply:
yes there is one kyma per restgetter, but the restgetter has its exclusive cache as it not attached to HelmClient cache, or maybe I didn't understand your point..